packages/coil-extension/src/popup/context/tipContext.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { createContext, useContext, useState, useMemo } from 'react'
+import { StorageService } from '@webmonetization/wext/services'
+
+//
+// Models
+//
+interface ITipContext {
+  currentTipAmount: number
+  maxAllowableTipAmount: number // possibly move to background in formatTipSettings util when done
+  setCurrentTipAmount: (amount: number) => void
+}
+
+interface ITipProvider {
+  storage: Pick<StorageService, 'get'>
+}
+
+// Context
+const TipContext = createContext({} as ITipContext)
+
+// Provider
+export const TipProvider: React.FC<ITipProvider> = props => {
+  const { storage, children } = props
+  const userObject = storage.get('user')
+
+  // todo: need to add tipCredits and paymentMethods to background queries
+  // todo: when nathan and brandon are done with COIL-1642 and COIL-1674
+  const {
+    tipSettings: {
+      lastTippedAmountUSD = 1,
+      inTippingBeta = false,
+      remainingDailyAmount = 0
+    } = {},
+    paymentMethods,
+    tipCredits = 10
+  } = userObject ?? {}
+
+  const creditCard = useMemo(() => {
+    return paymentMethods?.find((method: any) => {
+      if (method?.type === 'stripe') {
+        return method
+      }
+    })
+  }, [paymentMethods])
+
+  // todo: this could be a util instead of storing in state if initializing with the correct values doesn't work
+  // todo: possibly move to background in formatTipSettings util when done
+  const calculateMax = () => {
+    let newMax = 0
+    if (inTippingBeta == true) {
```

Comment:
Need to open a ticket to remove this code when the tipping beta is over.

Reply:
done -> COIL-1677 👍